src/app/AiPlayer.ts

```diff
@@ -0,0 +1,195 @@
+import { AnimalNames } from '~src/Enums/AnimalNamesEnum';
+import { AnimalRoles } from '~src/Enums/AnimalRolesEnum';
+import { Player } from '../Player';
+import { GameController } from './GameController';
+import { Offer } from './Trade';
+import { ConvertAnimalName } from './utils/ConvertAnimalName';
+
+export class AiPlayer extends Player {
+  /**
+   * Makes a move for AI Player
+   * @param gameController for the game that the player is participating in
+   */
+  makeAMove(gameController: GameController): void {
+    this.trade(gameController);
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+    gameController.breed();
+    if (gameController.theGameProcessor.checkWin()) {
+      gameController.checkIfGameIsWon();
+      return;
+    }
+  }
+  private trade(gameController: GameController): void {
+    //Try to win the game by trading one to many if possible
+    const oneToManyTrades: [AnimalNames, Offer[]][] = [
+      [
+        AnimalNames.HORSE,
+        [
+          [AnimalNames.COW, 1],
+          [AnimalNames.PIG, 1],
+          [AnimalNames.SHEEP, 3],
+          [AnimalNames.RABBIT, 6],
+        ],
+      ],
+      [
+        AnimalNames.COW,
+        [
+          [AnimalNames.PIG, 1],
+          [AnimalNames.SHEEP, 3],
+          [AnimalNames.RABBIT, 6],
+        ],
+      ],
+      [
+        AnimalNames.PIG,
+        [
+          [AnimalNames.SHEEP, 1],
+          [AnimalNames.RABBIT, 6],
+        ],
+      ],
+      [AnimalNames.SHEEP, [[AnimalNames.RABBIT, 6]]],
+    ];
+    for (const [animalToSell, target] of oneToManyTrades) {
+      if (this.herd.getAnimalNumber(animalToSell) > 1) {
+        const offer: Offer[] = [[animalToSell, 1]];
+        if (
+          gameController.theGame.theTrade.processOffer(
+            offer,
+            this,
+            target,
+          )
+        ) {
+          return;
+        }
+      } else if (this.herd.getAnimalNumber(animalToSell) === 0) {
+        break;
+      }
```

Comment:
Perhaps if you inverted the order you could do early break without he else if? 
`if (this.herd.getAnimalNumber(animalToSell) === 0) break`
`for (const [animalToSell, target] of oneToManyTrades) {`
      `if (this.herd.getAnimalNumber(animalToSell) > 1) {...`